packages/amplify-graphql-index-transformer/src/resolvers/resolvers.ts

```diff
@@ -335,6 +339,11 @@ export function validateSortDirectionInput(config: PrimaryKeyDirectiveConfigurat
  */
 export function appendSecondaryIndex(config: IndexDirectiveConfiguration, ctx: TransformerContextProvider): void {
   const { name, object, primaryKeyField } = config;
+  const dbType = getDBType(ctx, object.name.value);
+  if (dbType === 'MySQL') {
+    return;
```

Comment:
can we move this whole method to DDB VTL generator? Since it isn't doing anything for RDS data sources.

Reply:
This is not logically part of VTL generation. This must be a resource provider as it adds CFN resources but I don't see the value of doing to major refactor just to avoid a IF block.